plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/connection/OAuthDetailView.java

```diff
@@ -125,12 +117,15 @@ public Composite createControls(Composite parent, Object context, DataBindingCon
 		GridLayoutFactory.fillDefaults()
 				.numColumns(2).spacing(10, 10).applyTo(composite);
 		
-		tokenRequestLink = new Link(composite, SWT.WRAP);
+		StyledText tokenRequestLink = new StyledText(composite, SWT.WRAP);
+		StyledTextUtils.setTransparent(tokenRequestLink);
+		StyledTextUtils.setLinkText(MSG_TOKEN, tokenRequestLink);
 		GridDataFactory.fillDefaults()
 			.align(SWT.LEFT, SWT.CENTER).span(3, 1).applyTo(tokenRequestLink);
-		tokenRequestLink.setText(MSG_TOKEN);
-		tokenRequestLink.addSelectionListener(new AuthLinkHandler(parent.getShell(), authDetails));
-		
+		if(authDetails != null) {
+			authDetails.getRequestTokenLink();
+		}
```

Comment:
this was taken from the contructor in the SelectionAdapter. Is this correct? What is this retrieval all about? It seems to be "null" anyhow?